tests/unit/test_hnsw.cpp

```diff
@@ -1838,103 +1840,182 @@ TYPED_TEST(HNSWTest, rangeQueryCosine) {
-    auto res = serializer.checkIntegrity();
-    ASSERT_TRUE(res.valid_state);
+    for (size_t i = 0; i < 2; ++i) {
+        // Set index type.
+        params.multi = is_multi[i];
 
-    for (size_t i = 0; i < n; i++) {
-        float f[dim];
-        for (size_t j = 0; j < dim; j++) {
-            f[j] = (float)i;
-        }
-        VecSimIndex_AddVector(index, (const void *)f, i);
-    }
-    // Get index info and copy it, so it will be available after the index is deleted.
-    VecSimIndexInfo info = VecSimIndex_Info(index);
+        // Generate and add vectors to an index.
+        VecSimIndex *index = this->CreateNewIndex(params, is_multi[i]);
+        HNSWIndex<TEST_DATA_T, TEST_DIST_T> *hnsw_index = this->CastToHNSW(index);
 
-    // Persist index with the serializer, and delete it.
-    serializer.saveIndex(file_name);
-    VecSimIndex_Free(index);
+        std::vector<TEST_DATA_T> data(n * dim);
+        std::mt19937 rng;
+        rng.seed(47);
+        std::uniform_real_distribution<> distrib;
+        for (size_t i = 0; i < n * dim; ++i) {
+            data[i] = (TEST_DATA_T)distrib(rng);
+        }
+        for (size_t j = 0; j < n; ++j) {
+            VecSimIndex_AddVector(index, data.data() + dim * j, j % n_labels[i]);
+        }
 
-    // Create new index, set it into the serializer and extract the data to it.
-    auto new_index = test_utils::CreateNewIndex(params, VecSimType_FLOAT32);
-    ASSERT_EQ(VecSimIndex_IndexSize(new_index), 0);
-
-    serializer.reset(reinterpret_cast<HNSWIndex<float, float> *>(new_index));
-    serializer.loadIndex(file_name);
-
-    // Validate that the new loaded index has the same meta-data as the original.
-    VecSimIndexInfo new_info = VecSimIndex_Info(new_index);
-    ASSERT_EQ(info.algo, new_info.algo);
-    ASSERT_EQ(info.hnswInfo.M, new_info.hnswInfo.M);
-    ASSERT_EQ(info.hnswInfo.efConstruction, new_info.hnswInfo.efConstruction);
-    ASSERT_EQ(info.hnswInfo.efRuntime, new_info.hnswInfo.efRuntime);
-    ASSERT_EQ(info.hnswInfo.indexSize, new_info.hnswInfo.indexSize);
-    ASSERT_EQ(info.hnswInfo.max_level, new_info.hnswInfo.max_level);
-    ASSERT_EQ(info.hnswInfo.entrypoint, new_info.hnswInfo.entrypoint);
-    ASSERT_EQ(info.hnswInfo.metric, new_info.hnswInfo.metric);
-    ASSERT_EQ(info.hnswInfo.type, new_info.hnswInfo.type);
-    ASSERT_EQ(info.hnswInfo.dim, new_info.hnswInfo.dim);
-
-    res = serializer.checkIntegrity();
-    ASSERT_TRUE(res.valid_state);
-
-    // Add 1000 random vectors, override the existing ones to trigger deletions.
-    std::vector<float> data((n + 1) * dim);
-    std::mt19937 rng;
-    rng.seed(47);
-    std::uniform_real_distribution<> distrib;
-    for (size_t i = 0; i < (n + 1) * dim; ++i) {
-        data[i] = (float)distrib(rng);
-    }
-    for (size_t i = 0; i < n + 1; ++i) {
-        VecSimIndex_AddVector(new_index, data.data() + dim * i, i);
-    }
+        auto file_name = std::string(getenv("ROOT")) + "/tests/unit/data/1k-d4-L2-M8-ef_c10_" +
```

Comment:
couldn't find any function that is platform-independent to get pwd... (do you know any?)
and "."doesn't work because we are running tests from the dir
we define ROOT in the cmake file, right?
but yes anyway it won't work using the debugger unless you explicitly config ROOT

Reply:
`std::filesystem::current_path()` should work. Maybe use relative path to `data/`